packages/babel-parser/src/parser/statement.ts

```diff
@@ -464,14 +463,33 @@ export default abstract class StatementParser extends ExpressionParser {
 
       case tt._using:
         // using [no LineTerminator here] BindingList[+Using]
-        if (this.hasFollowingLineBreak()) {
+        if (
+          this.hasFollowingLineBreak() ||
+          this.state.containsEsc ||
+          !this.hasFollowingBindingAtom()
+        ) {
           break;
         }
-      // fall through
+        this.expectPlugin("explicitResourceManagement");
+        if (!allowDeclaration) {
+          this.raise(Errors.UnexpectedLexicalDeclaration, {
+            at: this.state.startLoc,
+          });
+        }
+        if (!this.scope.inModule && this.scope.inTopLevel) {
+          this.raise(Errors.UnexpectedUsingDeclaration, {
+            at: this.state.startLoc,
+          });
+        }
```

Comment:
Will this report a double error when parsing `label: using x = y` as a script?

Reply:
Good catch! In this case I lean to the `UnexpectedUsingDeclaration` because it's more specific than the other.